lib/pbench/agent/tool_data_sink.py

```diff
@@ -1259,9 +1267,16 @@ def execute(self):
             self._tm_tracking = self.record_tms(tms)
             self._num_tms = len(self._tm_tracking.keys())
 
-            # Tell the entity that started us who we are, indicating we're
-            # ready, and all the TMs are ready.
-            started_msg = dict(kind="ds", action="startup", status="success")
+            status = self.check_failed_tool_init(tms)
+            # Tell the entity that started us who we are, indicating if we
+            # are ready or not based on the status collected from TMs and
+            # proceed further accordingly.
+            if not status:
+                started_msg = dict(
+                    kind="ds", action="startup", tool_meister_info=tms, status="failed"
```

Comment:
I think this data is already stored as a key in Redis.

Reply:
When I first implemented the Tool Meister I was storing additional keys for each registered Tool Meister in Redis.  But that was removed mid-way through the development, if I recall correctly.

Having the full Tool Meister dictionary passed back through the response message works, but it might be better to have `record_tms()` also store the JSON blob in Redis as a key / value pair JSON dictionary entry so that it is also available for debugging.